golem_messages/message/base.py

```diff
@@ -297,6 +297,10 @@ def serialize(self, sign_func=None, encrypt_func=None):
 
         if sign_func is None:
             sign_func = _fake_sign
+        elif self.sig is not None:

```

Comment:
No, it will raise an exception only if you pass a different `sign_func` to the second call of `serialize().

Before this change if you had a message already signed with algorithm A and tried to `serialize()` it telling it to use algorithm B it would work without giving you an error even though the resulting message would not have a signature from algorithm B. It would fail later when you try to `deserialize()` it telling it to verify the signature using algorithm B.

Now it gives you an error immediately so you can't end up in that situation.